iodata/formats/qchemlog.py

```diff
@@ -0,0 +1,370 @@
+# IODATA is an input and output module for quantum chemistry.
+# Copyright (C) 2011-2019 The IODATA Development Team
+#
+# This file is part of IODATA.
+#
+# IODATA is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 3
+# of the License, or (at your option) any later version.
+#
+# IODATA is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, see <http://www.gnu.org/licenses/>
+# --
+"""Q-Chem Log file format.
+
+This module will load Q-Chem log file into IODATA.
+"""
+
+
+import re
+from typing import List, Tuple
+
+import numpy as np
+
+from ..docstrings import document_load_one
+from ..orbitals import MolecularOrbitals
+from ..periodic import sym2num
+from ..utils import LineIterator, kcalmol
+
+__all__ = []
+
+PATTERNS = ['*.qchemlog']
+
+
+@document_load_one("qchemlog", ['atcoords', 'athessian', 'atmasses', 'atnums', 'charge', 'energy',
+                                'g_rot', 'mo', 'lot', 'nelec', 'obasis_name', 'run_type', 'extra'])
+def load_one(lit: LineIterator) -> dict:
+    """Do not edit this docstring. It will be overwritten."""
+    data = load_qchemlog_low(lit)
+    result_labels = ['atcoords', 'atmasses', 'atnums', 'charge',
+                     'charge', 'energy', 'g_rot', 'run_type']
+    result = {label: data.get(label, None) for label in result_labels}
+    # build molecular orbitals
+    # todo: double check if this is right
+    # mo_energies
+    # # restricted case
+    # if not data['unrestricted']:
+    #     mo_energies = np.concatenate(
+    #         (data['alpha_mo_occupied'], data['alpha_mo_unoccupied']), axis=0)
+    #     mo_coeffs = np.empty((data['nbasis'], data['norba'])) * np.nan
+    #     mo_occs = np.zeros(mo_coeffs.shape[1]) * np.nan
```

Comment:
It seems that MO `coeffs` and `occs` are empty/zeros arrays; is it what you use to compute the number of electrons? If not, can you please explain what you get? The `coeffs` and `occs` don't exist in this Q-Chem log file.

Reply:
Yeah, `coeffs` and `occs` don't exist in qchem log file. I didn't use them to compute the number of electrons but adding number of alpha electrons and beta electrons, https://github.com/fwmeng88/iodata/blob/f0631ff3b76212877561da85aa48937afc08f978/iodata/formats/qchemlog.py#L68. 

The problem here is if I call `mol.charge` it will raise the wrong charge number with the above section uncommented.
```python
========================== test session starts ===========================
platform linux -- Python 3.7.7, pytest-5.0.1, py-1.8.1, pluggy-0.13.1
rootdir: /home/legend/Desktop/work/iodata_project/iodata
collected 2 items                                                        

iodata/test/test_qchemlog.py .F                                    [100%]

================================ FAILURES ================================
_________________________ test_load_one_qchemlog _________________________

    def test_load_one_qchemlog():
        with path('iodata.test.data', 'water_hf_ccpvtz_freq_qchem.out') as fn_qchemlog:
            mol = load_one(str(fn_qchemlog), fmt='qchemlog')
>       assert mol.charge == 0
E       AssertionError: assert nan == 0
E        +  where nan = IOData(atcharges={}, atcoords=array([[ 0.00575,  0.00426, -0.00301],\n       [ 0.27588,  0.88612,  0.25191],\n       [ 0... obasis_name='cc-pvtz', one_ints={}, one_rdms={}, run_type='freq', _spinpol=None, title=None, two_ints={}, two_rdms={}).charge

iodata/test/test_qchemlog.py:139: AssertionError
=================== 1 failed, 1 passed in 0.21 seconds ===================

```
Thank you.

Reply:
@fwmeng88, the test passes for me (and on Travis-CI as well). So, I cannot reproduce your problem.

Reply:
Problem fixed in cd5e515.